Paintroid/src/main/java/org/catrobat/paintroid/dialog/SaveInformationDialog.java

```diff
@@ -0,0 +1,203 @@
+		if (isStandard) {
+			FileIO.filename = "image";
+			FileIO.compressFormat = Bitmap.CompressFormat.JPEG;
+			FileIO.ending = ".jpg";
+			FileIO.compressQuality = 100;
+		}
+
+		if (FileIO.filename.equals("image")) {
+			this.setName = FileIO.filename + imageNumber;
+		} else {
+			this.setName = FileIO.filename;
+		}
+
+		this.permission = permissionCode;
+	}
+
+	@NonNull
+	@Override
+	@SuppressLint("InflateParams")
+	public Dialog onCreateDialog(Bundle savedInstanceState) {
+
+		inflater = getActivity().getLayoutInflater();
+		View customLayout = inflater.inflate(R.layout.dialog_pocketpaint_save, null);
+
+		AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+		builder.setView(customLayout);
+		AlertDialog dialog = builder.create();
+
+		initializeViews(customLayout);
+		initializeFunctioning();
+
+		if (FileIO.compressFormat == Bitmap.CompressFormat.PNG) {
+			mySpinner.setSelection(1);
+		} else {
+			mySpinner.setSelection(0);
+		}
+
+		return dialog;
+	}
+
+	private void initializeViews(View customLayout) {
+		specificFormatLayout = customLayout.findViewById(R.id.pocketpaint_save_format_specific_options);
+
+		jpgView = inflater.inflate(R.layout.dialog_pocketpaint_save_jpg_sub_dialog, specificFormatLayout, false);
+		mySeekbar = jpgView.findViewById(R.id.pocketpaint_jpg_seekbar_save_info);
+		percentage = jpgView.findViewById(R.id.pocketpaint_percentage_save_info);
+		mySpinner = customLayout.findViewById(R.id.pocketpaint_save_dialog_spinner);
+		saveButton = customLayout.findViewById(R.id.pocketpaint_save_information_save_button);
+		cancelButton = customLayout.findViewById(R.id.pocketpaint_save_information_cancel_button);
+		informationButton = customLayout.findViewById(R.id.pocketpaint_btn_save_info);
+		imageName = customLayout.findViewById(R.id.pocketpaint_image_name_save_text);
+	}
+
+	private void initializeFunctioning() {
+		List<String> spinnerArray = new ArrayList<>();
+		spinnerArray.add("jpg");
+		spinnerArray.add("png");
+
+		ArrayAdapter<String> adapter = new ArrayAdapter<>(mySpinner.getContext(),
+				android.R.layout.simple_spinner_item, spinnerArray);
+		adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+		mySpinner.setAdapter(adapter);
+
+		String percentageString = Integer.toString(FileIO.compressQuality);
+		percentage.setText(percentageString);
+		imageName.setText(setName);
+		mySeekbar.setProgress(FileIO.compressQuality);
+
+		mySeekbar.setOnSeekBarChangeListener(this);
+		mySpinner.setOnItemSelectedListener(this);
+
+		saveButton.setOnClickListener(new View.OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				FileIO.filename = imageName.getText().toString();
+				getPresenter().switchBetweenVersions(permission);
+				dismiss();
+			}
+		});
+
+		cancelButton.setOnClickListener(new View.OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				dismiss();
+			}
+		});
```

Comment:
Use setPositiveButton and setNegativeButton in with AlertDialog.Builder instead.